tests/data_context/abstract_data_context/test_data_docs_config_crud.py

```diff
@@ -0,0 +1,206 @@
+import copy
+from unittest import mock
+
+import pytest
+
+import great_expectations.exceptions as gx_exceptions
+from great_expectations.data_context import EphemeralDataContext
+
+
+@pytest.fixture
+def new_site_config() -> dict:
+    return {
+        "class_name": "SiteBuilder",
+        "module_name": "great_expectations.render.renderer.site_builder",
+        "store_backend": {
+            "module_name": "great_expectations.data_context.store.tuple_store_backend",
+            "class_name": "TupleFilesystemStoreBackend",
+            "base_directory": "/my_new_site/",
+        },
+        "site_index_builder": {"class_name": "DefaultSiteIndexBuilder"},
+    }
+
+
+class TestAddDataDocsSite:
+    @pytest.mark.unit
+    def test_add_data_docs_site(
+        self,
+        ephemeral_context_with_defaults: EphemeralDataContext,
+        new_site_config: dict,
+    ):
+        # Add a new site
+        new_site_name = "my_new_site"
+        ephemeral_context_with_defaults.add_data_docs_site(
+            site_name=new_site_name, site_config=new_site_config
+        )
+
+        # Check that the new site is present
+        assert new_site_name in ephemeral_context_with_defaults.get_site_names()
+
+    @pytest.mark.unit
+    def test_add_data_docs_site_persists(
+        self,
+        ephemeral_context_with_defaults: EphemeralDataContext,
+        new_site_config: dict,
+    ):
+        new_site_name = "my_new_site"
+        with mock.patch(
+            "great_expectations.data_context.EphemeralDataContext._save_project_config"
```

Comment:
Do we need to mock this? What happens if it gets called on an in-memory context?

Reply:
Mostly mocking just to make sure the method to persist gets called (`_save_project_config()`). We test the actual add in `test_add_data_docs_site` above.